### DISCUSSION TITLE
Shortcuts for TextArea

### DISCUSSION THREAD

**Discussion by Tim-Cao** [Textualize/textual#3548] [Q&A] [answered]
What's the shortcuts of copy selected text for TextArea?

What's the shortcuts of undo for TextArea?

**Comment by davep** [comment, marked as answer]
As with all of our widgets, the bindings for TextArea are given in its documentation: https://textual.textualize.io/widgets/text_area/#bindings

Currently TextArea has no internal clipboard facility, and I believe undo support is planned (and designed into the core of the widget) but not yet implemented.

**Reply by Tim-Cao** [reply]
Do you mean TextArea doesn't support copy to clipboard yet, right?

**Reply by davep** [reply]
No part of Textual supports interaction with the host operating system's clipboard. I don't know of any plans to add that. Generally people have tended to use pyperclip to add such support to their applications.

**Reply by Tim-Cao** [reply]
I see. Thanks

**Reply by Tim-Cao** [reply]
See https://github.com/Textualize/textual/discussions/4051#discussioncomment-8189821
